COLLABORATOR_GUIDE.md

```diff
@@ -73,6 +73,70 @@ All pull requests that modify executable code should be subjected to
 continuous integration tests on the
 [project CI server](https://ci.nodejs.org/).
 
+## Semver-major Changes
+
+Changes to API arguments, return values, error handling, timing, and
+non-internal side effects must be handled as semver-major changes. These may be
+landed in master at any time with sufficient CTC review and approval.
+
+If it becomes necessary to change an APIs arguments or return values (the
+API "signature"), the existing API *must* be Deprecated *first* and the new
+API either introduced in parallel or added after the next major Node.js version
+following the Deprecation as a replacement for the Deprecated API. In other
+words, existing API signatures *must not* change without a Deprecation.
+
+Adding or removing errors thrown or reported by an API, or altering the timing
+and non-internal side effects of the API can be done *without* a Deprecation
+cycle but *must* be landed as semver-major changes. For now, this also includes
+all changes to error messages.
+
+Changes that add new events to EventEmitter implementations may be landed as
+semver-minor changes. *Removing* events, or altering the semantics of existing
+events *must* be landed as semver-major and *must* go through a Deprecation
+cycle.
+
+## Deprecations
+
+Deprecation refers to the identification of APIs that should no longer be used
+and that will no longer be supported in Node.js.
+
+Node.js uses three fundamental Deprecation levels:
+
+* Soft-Deprecation (or Docs-only deprecation) refers to elements of the Node.js
+  public API that are being staged for deprecation in a future Node.js major.
+  An explicit notice indicating the deprecated status is added to the API
+  documentation *but no functional changes are implemented in the code* (there
+  will be no runtime deprecation warning).
+
+* Hard-Deprecation refers to the use of warnings or errors being reported at
+  runtime when a deprecated API is used. By default, Node.js will print a
+  warning to stderr the first time a deprecated API is used. A command-line
+  switch can be used to escalate such warnings into runtime errors that will
+  cause the Node.js process to exit. As with Soft-Deprecation, the documentation
+  for the API must be updated to clearly indicate the deprecated status.
+
+* End-of-life refers to APIs that have gone through Hard-Deprecation and are
+  ready to be removed from Node.js entirely.
+
+All Deprecations *must* be handled as semver-major changes and are subject to
+review and approval by the CTC. This means that while a PR adding a Deprecation
+can be landed in master at any time with appropriate review and approval, such
+changes are only permitted to be delivered in new major releases.
+
+A "Deprecation cycle" is one full Node.js major release.
```

Comment:
Nit: "..one full Node.js major release in one of the three Deprecation levels." This is something that logically follows from the text (advancement to any Deprecation level is semver-major, so it can only happen at the start of a major release, and it cannot change again until the next major releease) but it is good to spell out.